src/AddIns/Uno.UI.MediaPlayer.Skia.Gtk/GTKMediaPlayer.cs

```diff
@@ -0,0 +1,461 @@
+			}
+			_isPlaying = _mediaPlayer.State == VLCState.Playing;
+			_currentPositionBeforeFullscreenChange = _mediaPlayer.Time;
+			if (_videoView != null && _mediaPlayer != null)
+			{
+				Stop();
+				_ = Dispatcher.RunAsync(CoreDispatcherPriority.Normal, async () =>
+				{
+					await Initialized();
+					UpdateMedia();
+					_ = Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
+					{
+						UpdateVideoStretch();
+						Play();
+						Pause();
+						CurrentPosition = (double)_currentPositionBeforeFullscreenChange;
+						if (_isPlaying)
+						{
+							Play();
+						}
+					});
+				});
+			}
+		}
+	}
+
+	public void RequestFullScreen()
+	{
+		if (_mediaPlayer != null)
+		{
+			if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+			{
+				this.Log().Debug("RequestFullScreen");
+			}
+			_isPlaying = _mediaPlayer.State == VLCState.Playing;
+			_currentPositionBeforeFullscreenChange = _mediaPlayer.Time;
+
+			if (_videoView != null && _mediaPlayer != null)
+			{
+				Stop();
+				_ = Dispatcher.RunAsync(CoreDispatcherPriority.Normal, async () =>
+				{
+					await Initialized();
+					UpdateMedia();
+					_ = Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
+					{
+						UpdateVideoStretch();
+						Play();
+						Pause();
+						CurrentPosition = (double)_currentPositionBeforeFullscreenChange;
+						if (_isPlaying)
+						{
+							Play();
+						}
+					});
+				});
+			}
+		}
+	}
+
+	public void Stop()
+	{
+		if (_videoView != null && _mediaPlayer != null)
+		{
+			if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+			{
+				this.Log().Debug("Stop");
+			}
+			_mediaPlayer.Stop();
+			_videoView.Visible = false;
+		}
+	}
+
+	public void Pause()
+	{
+		if (_videoView != null && _mediaPlayer != null)
+		{
+			if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+			{
+				this.Log().Debug("Pause");
+			}
+			_mediaPlayer.Pause();
+			_videoView.Visible = true;
+		}
+	}
+
+	public void SetVolume(int volume)
+	{
+		if (_videoView != null && _mediaPlayer != null)
+		{
+			if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+			{
+				this.Log().Debug($"SetVolume ({volume})");
+			}
+			_mediaPlayer.Volume = volume;
+			_videoView.Visible = true;
+		}
+	}
+
+	public void Mute(bool IsMuted)
+	{
+		if (_videoView != null && _mediaPlayer != null)
+		{
+			if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+			{
+				this.Log().Debug($"Mute {IsMuted}");
+			}
+			_mediaPlayer.Mute = IsMuted;
+		}
+	}
+
+	public string Source
+	{
+		get => (string)GetValue(SourceProperty);
+		set => SetValue(SourceProperty, value);
+	}
+
+	public static DependencyProperty SourceProperty { get; } = DependencyProperty.Register(
+		"Source", typeof(string), typeof(GtkMediaPlayer), new PropertyMetadata(default(string),
+			OnSourceChanged));
+
+	private void UpdateVideoStretch()
+	{
+		if (Interlocked.CompareExchange(ref _isUpdating, 1, 0) == 1)
```

Comment:
Nick did it that way to improve performance and avoid UpdateVideoStretch being called repeatedly, since it is called every time the tranport control show/hide and the player is resized .
Which worked and greatly improved the method calls.

Reply:
My question is really to understand if some parts of this class is not running on the UI thread, for instance if we have some callbacks from VLC which are invoked from another thread. It's matter to understand if we have to use the `Interlocked` and other sync primitives while working in this class, or if was just a "reflex" to use it here (which is also valid).